mesonbuild/linkers/linkers.py

```diff
@@ -702,7 +702,7 @@ def get_soname_args(self, env: 'Environment', prefix: str, shlib_name: str,
             # For PE/COFF the soname argument has no effect
             return []
         sostr = '' if soversion is None else '.' + soversion
-        return self._apply_prefix(f'-soname,{prefix}{shlib_name}.{suffix}{sostr}')
+        return self._apply_prefix(['-soname', f'{prefix}{shlib_name}.{suffix}{sostr}'])
```

Comment:
Per IRC discussion, I was a bit disappointed that this seems to mean we would translate `-Wl,-soname,libfoo.so.1` instead into `-Wl,-soname -Wl,libfoo.so.1`

@dblsaiko suggested we could take varargs in the form

```python
self._apply_prefix(['-soname', 'libfoo.so.1], [-rpath', '/blah'])
```

to allow grouping arguments with commas where supported by the compiler-as-linker-driver, while letting the compiler also set a property signaling that it does need each individual argument passed with a separate -Wl.